test/extensions/filters/network/local_ratelimit/local_ratelimit_fuzz_test.cc

```diff
@@ -0,0 +1,96 @@
+#include "envoy/extensions/filters/network/local_ratelimit/v3/local_rate_limit.pb.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/stats/isolated_store_impl.h"
+
+#include "extensions/filters/network/local_ratelimit/local_ratelimit.h"
+
+#include "test/extensions/filters/network/local_ratelimit/local_ratelimit_fuzz.pb.h"
+#include "test/extensions/filters/network/local_ratelimit/local_ratelimit_fuzz.pb.validate.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/runtime/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::NiceMock;
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace LocalRateLimitFilter {
+struct ActiveFilter {
+  ActiveFilter(const ConfigSharedPtr& config) : filter_(config) {
+    filter_.initializeReadFilterCallbacks(read_filter_callbacks_);
+  }
+
+  NiceMock<Network::MockReadFilterCallbacks> read_filter_callbacks_;
+  Filter filter_;
+};
+
+DEFINE_PROTO_FUZZER(
+    const envoy::extensions::filters::network::local_ratelimit::LocalRateLimitTestCase& input) {
+
+  try {
+    TestUtility::validate(input);
+  } catch (const ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "ProtoValidationException: {}", e.what());
+    return;
+  } catch (const ProtobufMessage::DeprecatedProtoFieldException& e) {
+    ENVOY_LOG_MISC(debug, "DeprecatedProtoFieldException: {}", e.what());
+    return;
+  }
+  if (input.config().token_bucket().fill_interval().nanos() < 0) {
+    ENVOY_LOG_MISC(debug, "In fill_interval, nanos should not be negative!");
+    return;
+  }
+
+  try {
+    NiceMock<Event::MockDispatcher> dispatcher_;
+    Stats::IsolatedStoreImpl stats_store_;
+    NiceMock<Runtime::MockLoader> runtime_;
+    Event::MockTimer* fill_timer_ = new Event::MockTimer(&dispatcher_);
```

Comment:
If this is the only place we're doing active mocking, it might be possible to make this faster by using a fake. The other mocks, like dispatcher/runtime, could probably be static.

Reply:
> If this is the only place we're doing active mocking, it might be possible to make this faster by using a fake. The other mocks, like dispatcher/runtime, could probably be static.

In the previous commit, two objects have "EXPECTED_CALL": fill_timer_ and dispatcher_. 
fill_timer_ is used to "advance system time",
and dispatcher_'s EXPECTED_CALL is defined inside the constructor of MockTimer.

Just now I made dispatcher/runtime to be static.

I'm not sure on which object by "using a fake" here. Could you please explain further? Thanks for your suggestions! Those helped a lot!

Reply:
I'm suggesting it might be possible to have a fake dispatcher/timer for just this purpose, but it's probably more work than needed. It would be faster. Let's skip for now.